src/loaders/JsonImageInfoLoader.ts

```diff
@@ -231,60 +249,62 @@ class JsonImageInfoLoader implements IVolumeLoader {
       // using Image is just a trick to download the bits as a png.
       // the Image will never be used again.
       const img: HTMLImageElement = new Image();
-      img.onerror = function () {
+
+      img.onerror = () => {
         console.log("ERROR LOADING " + url);
       };
-      img.onload = (function (thisbatch) {
-        return function (event: Event) {
-          //console.log("GOT ch " + me.src);
-          // extract pixels by drawing to canvas
-          const canvas = document.createElement("canvas");
-          // nice thing about this is i could downsample here
-          const w = Math.floor((event?.target as HTMLImageElement).naturalWidth);
-          const h = Math.floor((event?.target as HTMLImageElement).naturalHeight);
-          canvas.setAttribute("width", "" + w);
-          canvas.setAttribute("height", "" + h);
-          const ctx = canvas.getContext("2d");
-          if (!ctx) {
-            console.log("Error creating canvas 2d context for " + url);
-            return;
-          }
-          ctx.globalCompositeOperation = "copy";
-          ctx.globalAlpha = 1.0;
-          ctx.drawImage(event?.target as CanvasImageSource, 0, 0, w, h);
-          // getImageData returns rgba.
-          // optimize: collapse rgba to single channel arrays
-          const iData = ctx.getImageData(0, 0, w, h);
-
-          const channelsBits: Uint8Array[] = [];
-          // allocate channels in batch
-          for (let ch = 0; ch < Math.min(thisbatch.length, 4); ++ch) {
-            channelsBits.push(new Uint8Array(w * h));
-          }
-          // extract the data
-          for (let j = 0; j < Math.min(thisbatch.length, 4); ++j) {
-            for (let px = 0; px < w * h; px++) {
-              channelsBits[j][px] = iData.data[px * 4 + j];
-            }
+
+      img.onload = (event) => {
```

Comment:
This is a big improvement!

Reply:
I (mercifully) never had a reason to use that make-a-function-that-returns-a-function-then-immediately-call-the-outer-function hack in the era before arrow functions, and I don't envy folks who had to write software professionally back when that was an accepted solution.